Automattic/jetpack
# Protect card: Add the number of Logins Blocked.

This PR populates the "Logins Blocked" section on the My Jetpack Protect card with the current number of logins that have been blocked by the Brute Force Protection module.

See Design P2: p1HpG7-rFA-p2 (Figma link is within the post)

Screenshots are shown below in the testing instructions.






## Proposed changes:

- Add the number of `blocked_logins` to the global `myJetpackInitialState` object.
- Output the `blocked_logins` to the "Logins Blocked" column on the Protect product card.

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-aP6-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






*** You can reference the link to the Figma design in the design P2 post: p1HpG7-rFA-p2

Testing prerequisites:
The only way I am aware to fully test this PR is to test with an already existing site (I tested with one of my own) that has Jetpack (or Protect) installed (with Brute Force Protection enabled (which is  default)) and it has been online for a period of time so that it naturally has had past blocked login attempts.
I am currently not aware of a way to manually/purposely force the Brute Force Protection module to block a login attempt (and therefore increment the total number of blocked logins) for testing purposes. I'll try to ask around to see if I can find a way to trigger a blocked login attempt.

One "sort-of" way to test this could be to manually update the 'wp option' where the blocked login count is updated every time a login is blocked.  Just so there is a number there, that is greater than 0.
To do this, run command:
`jetpack docker wp option update jetpack_protect_blocked_attempts 28` (or whatever number you want..)
Then follow the steps below.

- With a Jetpack site that has blocked login attempts, install the Jetpack Beta tester plugin and run this branch on either Jetpack or Protect or both.
    - Or of course you can run this PR branch on a Jurassic Ninja site (with the Jetpack Beta plugin), but the number of blocked logins will just be 0 in this case. (or you can update the 'wp option' as described above).
- Go to My Jetpack and look at the Protect product card.  Verify that the number of blocked logins is displayed in the "Logins Blocked" column.  If your site's number of blocked logins is > 0, then you can compare with the number of blocked logins that are displayed in the At a Glance section on the Jetpack Dashboard page.  See screenshot:

![Screen Shot on 2024-07-17 at 17-48-03](https://github.com/user-attachments/assets/1a3e7379-e67e-4b94-b515-290cd6b680a7)

.

![Screen Shot 2024-07-17 at 18 47 12](https://github.com/user-attachments/assets/da4f4e76-ff9d-4185-b46f-03837b9f809a)
